### ISSUE TITLE
FIRST/LAST_VALUE behavior changes

### ISSUE THREAD

**Issue by waynexia** [apache/datafusion#10732]
### Describe the bug

In #9960, `FIRST_VALUE` and `LAST_VALUE` are moved into another place, but that patch also changes the function signatures. They only accept `NUMERICS` after the PR. 

### To Reproduce

Try the following query in CLI:

d201ec742b08fa2a0549fc27f0b39622d67391ab

```sql
SELECT arrow_typeof(FIRST_VALUE('0.1'::DECIMAL(4,1)));
```

Result at d201ec742b08fa2a0549fc27f0b39622d67391ab (before #9960) is 
```sql
+----------------------------------------+
| arrow_typeof(FIRST_VALUE(Utf8("0.1"))) |
+----------------------------------------+
| Decimal128(4, 1)                       |
+----------------------------------------+
```
And after is
```sql
+----------------------------------------+
| arrow_typeof(first_value(Utf8("0.1"))) |
+----------------------------------------+
| Float64                                |
+----------------------------------------+
```

### Expected behavior

The output type changes because decimal is not one of `NUMERICS`. I can think of two points:
- Should decimal be one of `NUMERICS`?
- Should first/last value work like before?

My perspective is "yes" for both.

### Additional context

By the way, the function name also changes. From big letters to small letters.

**Comment by waynexia** [comment]
cc @jayzhan211. I'd like to make the corresponding change if the "expected behavior" looks reasonable.

**Comment by jayzhan211** [comment]
I had checked that this issue is solved in #10651, which I had added the UTF8 to the signature.

**Comment by jayzhan211** [comment]
Since #10651 is merged, I close the issue. if there is still any issue, we can reopen it. @waynexia
